### PR TITLE
GSdx OSD: Add Windows UI & small update

### LINKED ISSUE #1736: OSD Monitor doesn't update while fullscreen
***PCSX2 version:***
1.5.0.r4257.fce281473

***Plugins used:***
GSDX - OpenGL

***Description of the issue:***
With the OSD monitor enabled, the values update normally to copy what is in the titlebar.
Until you fullscreen the game window and the titlebar becomes hidden, then the monitor no longer updates.

***How to reproduce the issue:***
Check the 'Enable Monitor' checkbox in the OSD tab.
Run any game, and fullscreen the window

***PC specifications:***
OS: Arch Linux x86_64
GPU: Nvidia 770GT

Yay, first OSD bug report~!

### PR DESCRIPTION
**Pull request by FlatOutPS2**
Add Windows UI, improve/add a few log messages and add function to limit the amount of log messages on the screen at the same time/in the buffer.

Fixes #1736.

### CODE REVIEW

File: plugins/GSdx/GSSettingsDlg.h [hunk 1]
```diff
@@ -64,6 +64,20 @@ class GSHacksDlg : public GSDialog
 	}
 };
 
+class GSOSDDlg : public GSDialog
+{
+	int transparency;
```
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.h]: Is it necessary to make transparency a member variable?

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 2]
```diff
@@ -818,3 +822,112 @@ bool GSHacksDlg::OnMessage(UINT message, WPARAM wParam, LPARAM lParam)
...
+
+			theApp.SetConfig("osd_log_enabled", (int)IsDlgButtonChecked(m_hWnd, IDC_OSD_LOG));
+			theApp.SetConfig("osd_monitor_enabled", (int)IsDlgButtonChecked(m_hWnd, IDC_OSD_MONITOR));
+			theApp.SetConfig("osd_indicator_enabled", (int)IsDlgButtonChecked(m_hWnd, IDC_OSD_INDICATOR));
+
+			theApp.SetConfig("osd_transparency", transparency);
+
+			// OSD Font
+			int length = ((int)SendMessage(GetDlgItem(m_hWnd, IDC_OSD_FONT_EDIT), WM_GETTEXTLENGTH, 0, 0)) + 1;
+			char *buffer = new char[length];
```
Comment (FlatOutPS2) [inline plugins/GSdx/GSSettingsDlg.cpp]: Sure, I just copied it from the hacks/shader dialogs. I'll update both if you want.

File: plugins/GSdx/GSSettingsDlg.h (hunk 1 above)
Comment (FlatOutPS2) [inline plugins/GSdx/GSSettingsDlg.h]: Same as above. I copied how it worked in the hacks/shader dialogs. It is used in all the GSOSDDlg functions to keep track of the setting, so it is convenient.

File: plugins/GSdx/GSSettingsDlg.cpp (hunk 2 above)
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.cpp]: > I'll update both if you want.

Definitely change this one, do the other one if you feel like it.

File: plugins/GSdx/GSSettingsDlg.h (hunk 1 above)
Comment (turtleli) [inline plugins/GSdx/GSSettingsDlg.h]: Well, old code isn't necessarily good, and I say that as the person who didn't use the unique_ptr above. ;)

Personally I'd prefer just using a local variable when necessary (same applies to the other dialog sliders), but you can leave it as it is (well, you should at least prefix it with a `m_` so I don't search the code looking for where the variable is declared).

File: plugins/GSdx/GSSetting.cpp [hunk 3]
```diff
@@ -135,6 +139,10 @@ const char* dialog_message(int ID, bool* updateText) {
 			return "Makes textures partially or fully transparent as required by emulation. May cause unusual slowdowns for some games.";
 		case IDC_LOGZ:
 			return "Treat depth as logarithmic instead of linear. Recommended setting is on unless it causes graphical glitches.";
+		case IDC_OSD_MAX_LOG_EDIT:
+		case IDC_OSD_MAX_LOG:
+			return "Sets the maximum number of log messages on the screen or in the buffer at the same time. \n\n"
```
Comment (turtleli) [inline plugins/GSdx/GSSetting.cpp]: Spaces before \n is unnecessary.

File: plugins/GSdx/GSSetting.cpp [hunk 4]
```diff
@@ -90,6 +90,10 @@ const char* dialog_message(int ID, bool* updateText) {
 			return "Offset for the ST/UV texture coordinates. Fixes some odd texture issues and might fix some post processing alignment too.\n\n"
 				"  0500 0500, fixes Persona 3 minimap, helps Haunting Ground.\n"
 				"  0000 1000, fixes Xenosaga hair edges (DX10+ Issue)";
+		case IDC_OSD_LOG:
+			return "Prints log messages from the Function keys onscreen.";
+		case IDC_OSD_MONITOR:
+			return "Continuously prints the FPS and the EE, GS and UI percentages onscreen.";
```
Comment (turtleli) [inline plugins/GSdx/GSSetting.cpp]: UI isn't printed anymore (at least not for a typical user).

File: plugins/GSdx/GSdx.rc [hunk 5]
```diff
@@ -177,6 +177,33 @@ BEGIN
...
+    CONTROL         "Enable Log",IDC_OSD_LOG,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,15,6,54,10
+    CONTROL         "Enable Monitor",IDC_OSD_MONITOR,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,15,22,90,10
+    CONTROL         "Enable Indicator",IDC_OSD_INDICATOR,"Button",BS_AUTOCHECKBOX | WS_TABSTOP,15,38,90,10
+    LTEXT           "Character size:",IDC_STATIC,15,57,120,8
+    EDITTEXT        IDC_OSD_SIZE_EDIT,15,68,58,14,ES_RIGHT | ES_AUTOHSCROLL
+    CONTROL         "",IDC_OSD_SIZE,"msctls_updown32",UDS_SETBUDDYINT | UDS_ALIGNRIGHT | UDS_AUTOBUDDY | UDS_ARROWKEYS | UDS_NOTHOUSANDS,15,69,11,14
+    LTEXT           "Scrolling speed:",IDC_STATIC,15,90,120,8
+    EDITTEXT        IDC_OSD_SPEED_EDIT,15,101,58,14,ES_RIGHT | ES_AUTOHSCROLL
+    CONTROL         "", IDC_OSD_SPEED,"msctls_updown32", UDS_SETBUDDYINT | UDS_ALIGNRIGHT | UDS_AUTOBUDDY | UDS_ARROWKEYS | UDS_NOTHOUSANDS,15,102,11,14
+    LTEXT           "Maximum number of onscreen messages:",IDC_STATIC,15,121,150,8
```
Comment (turtleli) [inline plugins/GSdx/GSdx.rc]: `Maximum onscreen messages` could also work. I'm not fussed about this one though.

File: plugins/GSdx/GSOsdManager.cpp [hunk 6]
```diff
@@ -63,7 +64,8 @@ GSOsdManager::GSOsdManager() : m_atlas_h(0)
 	m_monitor_enabled = theApp.GetConfigB("osd_monitor_enabled");
 	m_indicator_enabled = theApp.GetConfigB("osd_indicator_enabled");
 	m_osd_transparency = theApp.GetConfigI("osd_transparency");
-	m_osd_transparency = m_osd_transparency < 0 ? 0 : m_osd_transparency > 200 ? 200 : m_osd_transparency;
+	m_osd_transparency = m_osd_transparency < 0 ? 0 : m_osd_transparency > 100 ? 100 : m_osd_transparency;
```
Comment (turtleli) [inline plugins/GSdx/GSOsdManager.cpp]: Alternative:
`m_osd_transparency = std::max(0, std::min(theApp.GetConfigI("osd_transparency"), 100));`

File: plugins/GSdx/GSdx.rc [hunk 7]
```diff
@@ -177,6 +177,33 @@ BEGIN
...
+    EDITTEXT        IDC_OSD_MAX_LOG_EDIT,15,132,58,14,ES_RIGHT | ES_AUTOHSCROLL
+    CONTROL         "", IDC_OSD_MAX_LOG,"msctls_updown32", UDS_SETBUDDYINT | UDS_ALIGNRIGHT | UDS_AUTOBUDDY | UDS_ARROWKEYS | UDS_NOTHOUSANDS,15,133,11,14
+    LTEXT           "Transparency:",IDC_STATIC,15,151,50,8	
+    CONTROL         "",IDC_OSD_TRANSPARENCY_SLIDER,"msctls_trackbar32",TBS_BOTH | TBS_NOTICKS | WS_TABSTOP,15,163,115,15
+    RTEXT           "100",IDC_OSD_TRANSPARENCY_TEXT,130,167,15,8
+    LTEXT           "Font:",IDC_OSD_FONT_TEXT,15,180,75,8
```
Comment (turtleli) [inline plugins/GSdx/GSdx.rc]: IDC_OSD_FONT_TEXT doesn't seem to be used for anything. You could just use IDC_STATIC unless you plan to do something with it.

Apart from that, it seems fine, so squash the fixup into the appropriate commit. I'll merge after a few days if people are generally happy (or no one comments).

**Pull request merged**